lib/sensu/client/process.rb

```diff
@@ -455,27 +455,33 @@ def setup_standalone
       # Setup the Sensu client JSON socket, for external check result
       # input. By default, the client socket is bound to localhost on
       # TCP & UDP port 3030. The socket can be configured via the
-      # client definition, `:socket` with `:bind` and `:port`. The
-      # current instance of the Sensu logger, settings, and transport
+      # client definition, `:socket` with `:bind` and `:port`. Users can opt-out
+      # of using the TCP and UDP socket by setting `:enabled` to `false`.
+      # The current instance of the Sensu logger, settings, and transport
       # are passed to the socket handler, `Sensu::Client::Socket`. The
       # TCP socket server signature (Fixnum) and UDP connection object
       # are stored in `@sockets`, so that they can be managed
       # elsewhere, eg. `close_sockets()`.
       def setup_json_socket
         options = @settings[:client][:socket] || Hash.new
+        options[:enabled] ||= true
         options[:bind] ||= "127.0.0.1"
         options[:port] ||= 3030
-        @logger.debug("binding client tcp and udp sockets", :options => options)
-        @sockets << EM::start_server(options[:bind], options[:port], Socket) do |socket|
-          socket.logger = @logger
-          socket.settings = @settings
-          socket.transport = @transport
-        end
-        @sockets << EM::open_datagram_socket(options[:bind], options[:port], Socket) do |socket|
-          socket.logger = @logger
-          socket.settings = @settings
-          socket.transport = @transport
-          socket.protocol = :udp
+        if options[:enabled] || options[:enabled].nil
```

Comment:
This should be `.nil?` read below for more flavor

In ruby there is a convention when asking questions such as if something is `nil` to use a question mark in the function name. This provides more clarity on its use. Take for example:
```ruby
$ irb
irb(main):001:0> a = nil
=> nil
irb(main):003:0> b = a.nil?
=> true
irb(main):004:0> b = nil?
=> false
```

Reply:
Also since you defined a default of true above the check for `nil` becomes less important as the only way this logic would be needed if someone explicitly set their config to `nil`. I don't know why someone would do that as if they wanted to explicitly set then would set `(true|false)`.

Reply:
Ah cool, yeah the `?` syntax makes sense. Does it make sense for me to define the default or should I scrap that and check for it being `nil`? Personally, I'm thinking being completely explicit with the defaults makes sense but, I'm game for whatever is more standard here.

Reply:
commented elsewhere but I'd prefer explicit default and drop the conditional.